src/DynamoCoreWpf/ViewModels/Search/NodeAutoCompleteSearchViewModel.cs

```diff
@@ -87,8 +103,138 @@ internal IEnumerable<NodeSearchElement> GetMatchingNodes()
                 if (element.OutputParameters.Any(op => op == inputPortType))
                     elements.Add(element);
             }
+            elements.Sort(CompareNodeSearchElementByTypeDistance);
+            
+            return elements.OrderBy(x =>x.Group);
+        }
+
+        /// <summary>
+        /// Does typeb derive from typea
+        /// </summary>
+        /// <param name="typea"></param>
+        /// <param name="typeb"></param>
+        /// <param name="core"></param>
+        /// <returns></returns>
+        private bool DerivesFrom(string typea, string typeb, ProtoCore.Core core)
+        {
+            try
+            {
+                //TODO mirrors can be cached until new types are imported...
+                var mirror1 = new ClassMirror(typea, core);
+                var mirror2 = new ClassMirror(typeb, core);
 
-            return elements;
+
+                //TODO as we do this check we can cache the type distance...
+                if (mirror2.GetClassHierarchy().Any(x => x.ClassName == mirror1.ClassName))
```

Comment:
@mjkkirschner @aparajit-pratap Can you explain what is type distance for me? I interpretate as levels of inheritance?

Reply:
yes, distance between classes in the inheritance chain.